### ISSUE TITLE
`aria-allowed-role` rule false positive for `<img>` with no alt but with accessible name

### ISSUE THREAD

**Issue by kospl** [dequelabs/axe-core#3143]
Product: axe-core

Expectation: `<img>` with role `button` and accessible name is a valid element.

Actual: axe core 4.3.3 reports it as a violation of `aria-allowed-role` rule.

Motivation: as described in https://www.w3.org/TR/html-aria/#el-img-no-alt `<img>` with no alt but WITH accessible name may have same roles as https://www.w3.org/TR/html-aria/#el-img including `button`.

<pre><code>
axe-core version: 4.3.3

Browser and Assistive Technology versions: Chrome/92.0.4515.159

For Tooling issues:
- Platform:  Windows
</code></pre>

Repro minimal example:

    <img role="button" aria-label="Search">
<br>

    axe.run({
      runOnly: ['aria-allowed-role']
    }, function (err, results) {
      console.log(results.violations);
    });

Result (axe-core 4.3.3):

![image](https://user-images.githubusercontent.com/26386660/131654911-f0f0be1f-44b1-4532-b379-890b39f6f067.png)

**Comment by straker** [comment]
So an image without an `alt` attribute is considered as `role=presentation` and is not allowed to have any role. See https://www.w3.org/TR/html-aria/#el-img-empty-alt

**Comment by kospl** [comment]
Related code block for `noAlt` [removed about year ago](https://github.com/dequelabs/axe-core/commit/292866526fcddecc8236e2a9a8d796f3f80932b4#diff-71c6e1ff51ae19a73e38e88147c50e670e87b1293961bff4b9f6a21fe3b56609), not sure why, but I think it has something to do with this false positive.

**Comment by padmavemulapati** [comment]
Validated with the latest axe-core dev branch code base, 
Test snippet:
```
<img role="button" aria-label="Search">
  <img role="button" alt="foo" /> 
<img role="button" aria-label="foo" /> 
```
![image](https://user-images.githubusercontent.com/47409579/146204428-6c60b524-67d6-42ef-bec6-078d3700124b.png)

Not finding any failures.

**Comment by kospl** [comment]
Yeah, some work have been done, but still fix is absent in all axe tools other than dev, that's why this issue is still open.

**Comment by WilcoFiers** [comment]
Nope, sorry. Our policy is to close the issue after QA has confirmed the issue is resolved in the develop branch. @kospl, this fix will be in the next released, which its currently looking will be the 4.4 release on January 31st.
